### PR TITLE
[online-DPO] online dpo cleanups

### PR DESCRIPTION
**Pull request by kashif**
* renamed some configs 
* removed unused configs
* call reference and reward model with no_grads
* use the dropout flag
* added test

### CODE REVIEW

File: trl/trainer/online_dpo_trainer.py [hunk 1]
```diff
@@ -63,12 +63,11 @@ def __init__(
...
-        self.policy.generation_config.eos_token_id = (
-            None  # disable `pad_token_id` and `eos_token_id` because we just want to
-        )
-        self.policy.generation_config.pad_token_id = None  # generate tokens without truncation / padding
+        # disable `pad_token_id` and `eos_token_id` because we just want to
+        policy.generation_config.eos_token_id = None
+        # generate tokens without truncation / padding
+        policy.generation_config.pad_token_id = None
 
         self.ref_policy = ref_policy
```
Comment (edbeeching) [inline trl/trainer/online_dpo_trainer.py:72]: can we change this to `ref_model` in this refactor? Other leave it to another PR.

**Review by edbeeching — APPROVED** [review]
Thank @kashif , just one comment other LGTM.

File: examples/scripts/online_dpo.py [hunk 2]
```diff
@@ -96,8 +96,9 @@ def tokenize(element):
     if tokenizer.chat_template is None:
         tokenizer.chat_template = SIMPLE_QUERY_CHAT_TEMPLATE
     reward_model = AutoModelForSequenceClassification.from_pretrained(config.reward_model_path, num_labels=1)
-    ref_policy = AutoModelForCausalLM.from_pretrained(config.sft_model_path)
-    policy = AutoModelForCausalLM.from_pretrained(config.sft_model_path)
+    ref_model = AutoModelForCausalLM.from_pretrained(config.sft_model_path)
+    model = AutoModelForCausalLM.from_pretrained(model_config.model_name_or_path)
```
Comment (kashif) [inline examples/scripts/online_dpo.py:100]: @edbeeching i am using the `model_name_or_path` for the main model for which the `tokenizer` was initialized (rather than the `sft_model_path` which i believe was for the `ref_model`)

**Comment by kashif** [comment]
test failing due to un-related CI issue, 

```
[gw1] PASSED tests/test_online_dpo_trainer.py::TestOnlineDPOTrainer::test_online_dpo_trainer_training 
```
